src/solver/hydro/daily2/h2o2_j_resoudre_le_probleme_lineaire.cpp

```diff
@@ -46,46 +46,30 @@ extern "C"
 #define SNPRINTF snprintf
 #endif
 
-void H2O2_J_ResoudreLeProblemeLineaire(DONNEES_MENSUELLES_ETENDUES* DonneesMensuelles,
+void H2O2_J_ResoudreLeProblemeLineaire(DONNEES_MENSUELLES_ETENDUES& DonneesMensuelles,
                                        int NumeroDeProbleme)
 {
-    int Var;
-    double* pt;
-    char PremierPassage;
+    auto& ProblemeHydrauliqueEtendu = DonneesMensuelles.ProblemeHydrauliqueEtendu;
 
-    PROBLEME_HYDRAULIQUE_ETENDU* ProblemeHydrauliqueEtendu;
-    PROBLEME_LINEAIRE_ETENDU_PARTIE_VARIABLE* ProblemeLineaireEtenduPartieVariable;
-    PROBLEME_LINEAIRE_ETENDU_PARTIE_FIXE* ProblemeLineaireEtenduPartieFixe;
+    auto& ProblemeLineaireEtenduPartieVariable
+      = ProblemeHydrauliqueEtendu.ProblemeLineaireEtenduPartieVariable[NumeroDeProbleme];
+    auto& ProblemeLineaireEtenduPartieFixe
+      = ProblemeHydrauliqueEtendu.ProblemeLineaireEtenduPartieFixe[NumeroDeProbleme];
 
-    PROBLEME_SIMPLEXE* Probleme;
-    PROBLEME_SPX* ProbSpx;
+    auto ProbSpx = (PROBLEME_SPX*)ProblemeHydrauliqueEtendu.ProblemeSpx[NumeroDeProbleme];
+    auto Probleme = std::make_unique<PROBLEME_SIMPLEXE>();
 
-    PremierPassage = OUI;
-
-    ProblemeHydrauliqueEtendu = DonneesMensuelles->ProblemeHydrauliqueEtendu;
-
-    ProblemeLineaireEtenduPartieVariable
-      = ProblemeHydrauliqueEtendu->ProblemeLineaireEtenduPartieVariable[NumeroDeProbleme];
-    ProblemeLineaireEtenduPartieFixe
-      = ProblemeHydrauliqueEtendu->ProblemeLineaireEtenduPartieFixe[NumeroDeProbleme];
-
-    ProbSpx = (PROBLEME_SPX*)ProblemeHydrauliqueEtendu->ProblemeSpx[NumeroDeProbleme];
-
-    Probleme = (PROBLEME_SIMPLEXE*)ProblemeHydrauliqueEtendu->Probleme;
-    if (Probleme == NULL)
+    if (!Probleme)
```

Comment:
If you really want to check memory allocation, then you should `catch(std::bad_alloc)`. But there is already a global catch (src/solver/main.cpp:144), so we don't need it here.